pkg/context/context.go

```diff
@@ -18,12 +18,15 @@ type Context struct {
 	SnapshotPath      string
 	SnapshotRoot      string
 	SnapshotExclusive bool
+	SnapshotPreserve  bool
```

Comment:
bad rebase, will fix in the next upload which I need to do anyway to add tests.

Reply:
@fromanirh you still working on this patch, right?

Reply:
yes! totally ongoing, but not sure I can make it for end of the week (aka 0.7 cutoff).